### PR TITLE
Fix linking against Homebrew's Tcl/Tk 8.6.13 in MacOS

### PR DESCRIPTION
**Pull request by startergo**
Correct the path for `tcl-tk` `include` for homebrew

Make sure you have checked all steps below.

### Prerequisite
* [ ] Please consider implementing the feature as a hook script or plugin as a first step.
  * pyenv has some powerful support for plugins and hook scripts. Please refer to [Authoring plugins](https://github.com/pyenv/pyenv/wiki/Authoring-plugins) for details and try to implement it as a plugin if possible.
* [ ] Please consider contributing the patch upstream to [rbenv](https://github.com/rbenv/rbenv), since we have borrowed most of the code from that project.
  * We occasionally import the changes from rbenv. In general, you can expect changes made in rbenv will be imported to pyenv too, eventually.
  * Generally speaking, we prefer not to make changes in the core in order to keep compatibility with rbenv.
* [ ] My PR addresses the following pyenv issue (if any)
  - Closes https://github.com/pyenv/pyenv/issues/2794

### Description
- [x] Here are some details about my PR
https://github.com/pyenv/pyenv/issues/2794#issuecomment-1763182002
### Tests
- [x] My PR adds the following unit tests (if any)
```
pyenv install 2.7.18
python -m Tkinter
```

### CODE REVIEW

File: plugins/python-build/bin/python-build [hunk 1]
```diff
@@ -1645,7 +1645,7 @@ use_homebrew_tcltk() {
     if [[ -z "$PYTHON_BUILD_TCLTK_USE_PKGCONFIG" ]]; then
       local tcltk_version="$(sh -c '. '"$tcltk_libdir"'/lib/tclConfig.sh; echo $TCL_VERSION')"
       package_option python configure --with-tcltk-libs="-L$tcltk_libdir/lib -ltcl$tcltk_version -ltk$tcltk_version"
-      package_option python configure --with-tcltk-includes="-I$tcltk_libdir/include"
+      package_option python configure --with-tcltk-includes="-I$tcltk_libdir/include/tcl-tk"
```
Comment (native-api) [inline plugins/python-build/bin/python-build:1648]: This seems to only be true for the latest Homebrew package. So we can't apply this universally.
```
$ apt-file search /tcl.h
<...>
tcl8.6-dev: /usr/include/tcl8.6/tcl.h
```
```
% find / -name tcl.h 2>/dev/null
/usr/local/Cellar/tcl-tk/8.6.13_5/include/tcl-tk/tcl.h
<...>
/Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/tcl.h
/Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/System/Library/Frameworks/Tcl.framework/Versions/8.5/Headers/tcl.h
/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include/tcl.h
/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/Tcl.framework/Versions/8.5/Headers/tcl.h
```

**Review by native-api — CHANGES_REQUESTED** [review]

**Review by startergo — COMMENTED** [review]
Added conditional coding of `include` folder rather than hardcoded one

**Review by startergo — COMMENTED** [review]
`tcl.h ` can be in multiple locations. How can we ensure `tcl-tk` from Homebrew is used in the `use_homebrew_tcltk()`?

**Comment by native-api** [comment]
This still links 2.7.18 against 8.5 . [As I said, 2.7.18's Tkinter build is bugged](https://github.com/pyenv/pyenv/issues/2794#issuecomment-1727580772), it appends rather than prepends the required directory to the compiler search path -- so it doesn't pick the intended version when other directories or the compiler search path also have it.

Looks like to work around that, we'd need to include the mode to configure Tcl/Tk via `CPPFLAGS`/`LDFLAGS` rather than through `--with-tcltk-*`.

---

However.

I've looked through the history... and we seem to have never done this before! So, configuring 2.7 with Homebrew's Tcl/Tk never worked! I was suspecting that this worked before and then we broke it -- then this would be a bugfix.
But it's not. This'll be a new feature.

And because 2.7 is EOL, adding new features just for it would be against our [Deprecation policy](https://github.com/pyenv/pyenv/blob/master/CONTRIBUTING.md#deprecation-policy).

---

So I suggest that we should rather make a patch for 2.7.18 that would properly prepend the dirs to the compiler's search paths.

**Comment by native-api** [comment]
> And because 2.7 is EOL, adding new features just for it would be against our [Deprecation policy](https://github.com/pyenv/pyenv/blob/master/CONTRIBUTING.md#deprecation-policy).

That said, PyPy-2.7 is NOT EOL -- so it this feature applies to it as well, it's fine to add.

**Comment by startergo** [comment]
> This still links 2.7.18 against 8.5

True, but at least it is not borked. Interestingly if I hardcode it like this:
```
package_option python configure --with-tcltk-includes="-I$tcltk_libdir/include/tcl-tk"
```
it links `2.7.18` against `8.6`, but if I do conditional setup like this:
```
if [ -d "$tcltk_libdir/include/tcl-tk" ]; then
   package_option python configure --with-tcltk-includes="-I$tcltk_libdir/include/tcl-tk"
fi
package_option python configure --with-tcltk-includes="-I$tcltk_libdir/include"	
```
It links against `8.5`. Go figure...

**Comment by startergo** [comment]
> Looks like to work around that, we'd need to include the mode to configure Tcl/Tk via `CPPFLAGS`/`LDFLAGS` rather than through `--with-tcltk-*`.

Isn't it better to rely on `pkg-config` and include it in the `brew` requirement? At least it would use relative paths.

**Review by startergo — COMMENTED** [review]
aah! my bad...

**Comment by native-api** [comment]
Wow! I don't know how but it works.

```
% find ~/.pyenv/versions/2.7.18 -name _tkinter\* | xargs otool -L
/Users/admin/.pyenv/versions/2.7.18/lib/python2.7/lib-dynload/_tkinter.so:
	/usr/local/opt/tcl-tk/lib/libtcl8.6.dylib (compatibility version 8.6.0, current version 8.6.13)
	/usr/local/opt/tcl-tk/lib/libtk8.6.dylib (compatibility version 8.6.0, current version 8.6.13)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1319.0.0)
```

**Review by startergo — COMMENTED** [review]
Makes sense. Thanks.

**Comment by native-api** [comment]
Thanks to you, too! Way to prove me wrong!

**Pull request merged**